docs/installation_guide.md

```diff
@@ -1,126 +1,145 @@
-
-   **IMPORTANT:** When upgrading from an existing Harbor instance, you must make sure **auth_mode** is the same in ```harbor.cfg``` before launching the new version of Harbor. Otherwise, users 
-may not be able to log in after the upgrade.
-* **ldap_url**: The LDAP endpoint URL (e.g. `ldaps://ldap.mydomain.com`).  _Only used when **auth_mode** is set to *ldap_auth* ._    
-* **ldap_searchdn**: The DN of a user who has the permission to search an LDAP/AD server (e.g. `uid=admin,ou=people,dc=mydomain,dc=com`).
-* **ldap_search_pwd**: The password of the user specified by *ldap_searchdn*.
-* **ldap_basedn**: The base DN to look up a user, e.g. `ou=people,dc=mydomain,dc=com`.  _Only used when **auth_mode** is set to *ldap_auth* ._ 
-* **ldap_filter**: The search filter for looking up a user, e.g. `(objectClass=person)`.
-* **ldap_uid**: The attribute used to match a user during a LDAP search, it could be uid, cn, email or other attributes.
-* **ldap_scope**: The scope to search for a user, 0-LDAP_SCOPE_BASE, 1-LDAP_SCOPE_ONELEVEL, 2-LDAP_SCOPE_SUBTREE. Default is 2. 
-* **ldap_timeout**: Timeout (in seconds)  when connecting to an LDAP Server. Default is 5.
-* **ldap_verify_cert**: Verify certificate from LDAP server. Default is true.
-* **ldap_group_basedn**: The base dn from which to lookup a group in LDAP/AD, e.g. `ou=group,dc=mydomain,dc=com`.
-* **ldap_group_filter**: The filter to search LDAP/AD group, e.g. `objectclass=group`.
-* **ldap_group_gid**: The attribute used to name a LDAP/AD group, it could be cn, name.
-* **ldap_group_scope**: The scope to search for ldap groups. 0-LDAP_SCOPE_BASE, 1-LDAP_SCOPE_ONELEVEL, 2-LDAP_SCOPE_SUBTREE. Default is 2.
-* **self_registration**: (**on** or **off**. Default is **on**) Enable / Disable the ability for a user to register himself/herself. When disabled, new users can only be created by the Admin user, only an admin user can create new users in Harbor.  _NOTE: When **auth_mode** is set to **ldap_auth**, self-registration feature is **always** disabled, and this flag is ignored._  
-* **token_expiration**: The expiration time (in minutes) of a token created by token service, default is 30 minutes.
-* **project_creation_restriction**: The flag to control what users have permission to create projects.  By default everyone can create a project, set to "adminonly" such that only admin can create project.
+
+###### Email settings
+
+These parameters are needed for Harbor to be able to send a user a "password reset" email and are only necessary if that functionality is needed. Also note that by default SSL connectivity is _not_ enabled. If your SMTP server requires SSL but does _not_ support STARTTLS then you should enable SSL by setting `email_ssl = true`. Set `email_insecure = true` if the email server uses a self-signed or untrusted certificate. For a detailed discussion of "email_identity" please refer to [rfc2595](https://tools.ietf.org/rfc/rfc2595.txt)
+
+* email_server = smtp.mydomain.com
```

Comment:
I'd make these a table to be consistent with the rest of the parameters described in this section.